src/mmapbitarray.c

```diff
@@ -472,6 +487,16 @@ int _initialize_file(int fd, size_t end, BTYPE num_bits, const char * header, in
     return 0;
 }
 
+static void _initialize_in_memory_vector(DTYPE * vector, BTYPE num_bits, const char * header, int32_t header_len) {
+    char *dest = (char *)vector;
+    memcpy(dest, MBAMAGIC, MBAMAGICSIZE);
+    dest += MBAMAGICSIZE;
+    memcpy(dest, &num_bits, sizeof(BTYPE));
+    dest += sizeof(BTYPE);
+    memcpy(dest, &header_len, sizeof(header_len));
```

Comment:
Sorry, I didn't understand this comment :thinking:

Reply:
IIUC, this is writing a MmapBitArray structure into dest. But the structure has a different type for this field. With a different size. I think it only works because it's little-endian and there's a subsequent memcpy to copy the rest, but still it looks like something that needs fixing.

Reply:
Ohh, now I understand. No, here `dest` is the `MmapBitArray->vector` field, which in the MMaped version corresponds to the file in the disk. `header_len` here is actually `sizeof(BloomFilter)` that it is a `size_t`, but it's casted to `int32_t` implicitly by the compiler when `mbarray_Create_Malloc` is called. In theory it _could_ overflow, but the BloomFilter structure is small so it's not going to happen. Either way, the objective of this function is to be a `copy-paste` of `_initialize_file`, with the difference that instead of writing to a file descriptor we are just writing to an array. So instead of using `write` which automatically increases the offset , we have to use `memcpy` and do the increase by hand.

Reply:
:thinking: 

...I'll trust your tests...